Apply meta registrations, decomp table unconditionally

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #84310
* __->__ #84096
* #84090

Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.

eellison committed:
Apply meta registrations, decomp table unconditionally

[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

ezyang commented:
The main problem with doing this is you will no longer be routing to meta implementations generated by structured kernels, which are guaranteed to be accurate. But ofc if there are no bugs in primtorch this doesn't make a material difference lol.

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison commented:
> The main problem with doing this is you will no longer be routing to meta implementations generated by structured kernels, which are guaranteed to be accurate. But ofc if there are no bugs in primtorch this doesn't make a material difference lol.

I'm bailing on this temporarily because I can't figure out the SymInt errors.. but i'm at least unifying the SymInt/Fake paths

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]

eellison committed:
Update on "Apply meta registrations, decomp table unconditionally"


Apply meta registrations, decomp table unconditionally, not just for symbolic tensors. We should unify the paths concrete/symbolic fake tensors take so bugs occur consistently between the two.

Also does various cleanup.


[ghstack-poisoned]
